flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamJoinToCoProcessTranslator.scala

```diff
@@ -145,6 +145,7 @@ class DataStreamJoinToCoProcessTranslator(
           genFunction.name,
           genFunction.code,
           queryConfig)
+      case _ => throw new TableException(s"$joinType is not supported.")
```

Comment:
I think this one should throw `ValidationException` similar to other Translator classes.